int8  query_type    ### 0: perform soma object query
		    ### 1: return all object types and ids
		    ### 2: return min-max timestamp limits of the collection
		    
		    

bool uselowertime   ### use lower time limit as a query field
bool useuppertime   ### use upper time limit as a query field
bool usedates	    ### use dates as a query field. If the upper or lower date is 0, it is ignored.
bool useweekday     ### use weekday as a query field
bool useroi	    ### use roi as a query field


uint32 lowerhour     ### lower limit of the hour

uint32 upperhour     ### upper limit of the hour

uint32 lowerminutes  ### lower limit of minutes

uint32 upperminutes  ### upper limit of minutes

uint64 lowerdate     ### lower date in timestamp format

uint64 upperdate     ### upper date limit in timestamp format

uint8 weekday        ### iso weekday 1-Monday 7-Sunday

string[] objectids   ### array of object ids that will be queried with or statement

string[] objecttypes ### array of object types that will be queried with or statement   

string roi_id        ### id of the roi that will be used for within query

string config		   ### retrieve objects with specific configuration


---
int32[]  timedatelimits    ### timedatelimits[0] -> min-date in timestamp format
			   ### timedatelimits[1] -> max-date in timestamp format

string[] types             ### vector of object types present in current collection

string[] ids		   ### vector of object ids present in current collection

string[] unique_ids        ### vector of mongo unique ids of objects

soma_msgs/SOMAObject[] objects  ### vector of returned objects

string queryjson 	   ### performed query in json format
